cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -79,13 +87,53 @@
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  private static NamespaceId configuredNamespace = configureTestNamespace();
+  // boolean is true if we created the namespace; indicates that we are responsible for cleaning it
+  private static Map<NamespaceId, Boolean> registeredNamespaces = new HashMap<>();
+
   private AccessToken accessToken;
 
   @Before
   public void setUp() throws Exception {
     checkSystemServices();
     assertUnrecoverableResetEnabled();
-    assertIsClear();
+
+    boolean createdByUs = false;
+    if (!getNamespaceClient().exists(configuredNamespace.toId())) {
```

Comment:
https://github.com/caskdata/cdap/pull/6562/files/45c1c1512ecbd91f22c41cacd39f94b0ca910418#r76140496 seems like a fair argument. Should we document it in the README?